bolt-sidecar/src/api/commitments/firewall_recv.rs

```diff
@@ -0,0 +1,548 @@
+        api_events_rx
+    }
+
+    /// Returns the local addr the server is listening on (or configured with).
+    pub fn urls(&self) -> &[Url] {
+        self.urls.as_slice()
+    }
+}
+
+async fn handle_connection(
+    url: Url,
+    api_events_tx: mpsc::Sender<CommitmentEvent>,
+    ping_rx: broadcast::Receiver<()>,
+) {
+    // TODO: fill this
+    let ws_config = WebSocketConfig { ..Default::default() };
+
+    loop {
+        match connect_async_with_config(url.to_string(), Some(ws_config), false).await {
+            Ok((stream, response)) => {
+                info!(?url, ?response, "opened websocket connection");
+                let (write_sink, read_stream) = stream.split();
+
+                // For each opened connection, create a new commitment processor
+                // able to handle incoming message requests.
+                let commitment_request_processor = CommitmentRequestProcessor::new(
+                    url.clone(),
+                    api_events_tx.clone(),
+                    write_sink,
+                    read_stream,
+                    ping_rx.resubscribe(),
+                );
+                // Run the commitment processor indefinitely, reconnecting on failure.
+                commitment_request_processor.await
+            }
+            Err(e) => {
+                error!(?e, ?url, "failed to connect");
+            }
+        }
+
+        // Reconnect on failure
+        // TODO: add backoff
+        warn!(?url, "connection lost. reconnecting...");
+        tokio::time::sleep(Duration::from_secs(5)).await;
+    }
+}
+
+/// The [CommitmentRequestProcessor] handles incoming commitment requests a the websocket
+/// connection, and forwards them to the [CommitmentEvent] tx channel for processing.
+struct CommitmentRequestProcessor {
+    /// The URL of the connected websocket server.
+    url: Url,
+    /// The channel to send commitment events to be processed.
+    api_events_tx: mpsc::Sender<CommitmentEvent>,
+    /// The websocket writer sink.
+    write_sink: SplitSink<WebSocketStream<MaybeTlsStream<TcpStream>>, Message>,
+    /// The websocket reader stream.
+    read_stream: SplitStream<WebSocketStream<MaybeTlsStream<TcpStream>>>,
+    /// The receiver for keep-alive ping messages.
+    ping_rx: broadcast::Receiver<()>,
+    /// The collection of pending commitment requests responses, sent with [api_events_tx].
+    /// NOTE: Is there a better way to avoid this monster type?
+    /// SAFETY: the `poll` implementation of this struct promptly handles these responses and
+    /// ensures this vector doesn't grow indefinitely.
+    pending_commitment_responses: FuturesUnordered<
+        Pin<
+            Box<
+                dyn Future<
+                        Output = Result<
+                            Result<SignedCommitment, CommitmentError>,
+                            oneshot::error::RecvError,
+                        >,
+                    > + Send,
+            >,
+        >,
+    >,
```

Comment:
A good start can be do extract the type alias like
```rust
type PendingCommitmentResult = impl Future<Output = Result<SignedCommitment, CommitmentError>>;

pending_commitment_responses: FuturesUnordered<PendingCommitmentResult>
```

or similar (if you need pin<box as well)

Also I'm pretty sure we can get rid of the oneshot channel error and handle it in the poll implementation, as it can only happen when the sender is dropped